Documentation/project-docs/developer-guide.md

```diff
@@ -96,67 +112,61 @@ build-managed -debug -buildArch=x64
 ```
 
-- Building only the managed libraries with /flp:v=diag. It doesn't restore packages, build tests or build packages.
+##Build Tests
+To build the tests and run them you can call build-test script. The same parameters you pass to build should also be passed to build-tests script to ensure you are building and running the tests on the same configuration you have build the product on.
+
+**Examples**
+- The following shows how to build tests but not run them
 ```
-build-managed -binaries -verbose
+build-tests -skiptests
 ```
 
-- Building the managed libraries and the tests but not running tests.
+- The following builds and runs all tests for netcoreapp.
 ```
-build-managed -skiptests
+build-tests -debug -framework netcoreapp
 ```
 
 - The following example shows the argument `--`. Everything that is after it is not going to be processed and it is going to be passed as it is.
-Use it to pass extra msbuild properties.
+Use it to pass extra msbuild properties, in this case to ignore tests ignored in CI.
 ```
-build-managed -binaries -- /p:WithoutCategories=IgnoreForCI
+build-tests -- /p:WithoutCategories=IgnoreForCI
 ```
 
 ### Building individual CoreFx DLLs
 
+*Note*: Before working on individual projects or test projects you **must** run `build.cmd` once before beginning that work.
+
 Under the src directory is a set of directories, each of which represents a particular assembly in CoreFX.
-  
-For example the src\System.Diagnostics.DiagnosticSource directory holds the source code for the System.Diagnostics.DiagnosticSource.dll assembly. Each of these directories includes two projects, one for the DLL being built and one for the tests, both specified by a .builds file.
 
-You can build the DLL for System.Diagnostics.DiagnosticSource.dll by going to the src\System.Diagnostics.DiagnosticsSource\src directory and typing `msbuild System.Diagnostics.DiagnosticSource.builds`. The DLL ends up in bin\AnyOS.AnyCPU.Debug\System.Diagnostics.DiagnosticSource.
+For example the src\System.Diagnostics.DiagnosticSource directory holds the source code for the System.Diagnostics.DiagnosticSource.dll assembly. Each of these directories includes two projects, one for the DLL being built and one for the tests.
 
-You can build the tests for System.Diagnostics.DiagnosticSource.dll by going to 
-src\System.Diagnostics.DiagnosticSource\tests and typing `msbuild System.Diagnostics.DiagnosticSource.Tests.builds`.
+You can build the DLL for System.Diagnostics.DiagnosticSource.dll by going to the src\System.Diagnostics.DiagnosticsSource\src directory and typing `msbuild System.Diagnostics.DiagnosticSource.csproj`. The DLL ends up in bin\AnyOS.AnyCPU.Debug\System.Diagnostics.DiagnosticSource as well as bin\runtime\[BuildConfiguration].
+
+You can build the tests for System.Diagnostics.DiagnosticSource.dll by going to
+src\System.Diagnostics.DiagnosticSource\tests and typing `msbuild System.Diagnostics.DiagnosticSource.Tests.csproj`.
```

Comment:
I think it will be pretty random error message about missing references or missing types. I've created https://github.com/dotnet/corefx/issues/15121 to try and make that error cleaner.